scripts/evaluate.py

```diff
@@ -0,0 +1,110 @@
+import sys
+import os
+
+import torch
+import numpy as np
+from icecream import ic
+from dataclasses import dataclass
+import albumentations as A
+from tqdm import tqdm
+import cv2
+
+from .metrics import compute_average_distance, compute_accuracy
+from .utils import load_pretrained_weights, AverageMeter, mkdir_if_missing
+from dataloaders import Objectron
+
+module_path = os.path.abspath(os.path.join('/home/prokofiev/3D-object-recognition/3rdparty/Objectron'))
+if module_path not in sys.path:
+    sys.path.append(module_path)
+
+from objectron.dataset import graphics
+
+@dataclass
+class Tester:
+    model: object
+    test_loader: object
+    cfg: dict
+    checkpoint_path: str = ''
+    device: str = 'cuda'
+    samples: object = 'random'
+    num_samples: int = 10
+    path_to_save_imgs: str = './testing_images'
+
+    def evaluate(self):
+        ds = Objectron(self.cfg.data.root, mode='test')
+        mkdir_if_missing(self.path_to_save_imgs)
+        if self.samples == 'random':
+            indexes = np.random.choice(len(ds), self.num_samples, replace=False)
+        else:
+            assert isinstance(self.samples, list)
+            indexes = self.samples
+
+        for idx in tqdm(indexes):
+            # pull next data
+            prefetch_img, img, gt_kp, gt_cat = ds[idx]
+            # feed forward net and put it on cpu
+            pred_kp, pred_cat = self.model(torch.unsqueeze(img,0))
+            pred_kp = pred_kp.to('cpu')
```

Comment:
Let's mimic pytorch behavior: loss and similar functions don't force user to switch devices of the input tensors explicitly.